Client/Application/Delegates/AppDelegate.swift

```diff
@@ -260,9 +260,28 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
         }
         
         if let urp = UserReferralProgram.shared {
-            if isFirstLaunch {
-                let refCode = UserReferralProgram.sanitize(input: UIPasteboard.general.string)
-                if refCode != nil { UIPasteboard.general.clearPasteboard() }
+            if Preferences.URP.referralLookupOutstanding.value == nil {
+                // This preference has never been set, and this means it is a new or upgraded user.
+                // That distinction must be made to know if a network request for ref-code look up should be made.
+                
+                // Setting this to an explicit value so it will never get overwritten on subsequent launches.
+                // Upgrade users should not have ref code ping happening.
+                Preferences.URP.referralLookupOutstanding.value = isFirstLaunch
+            }
+            
+            if Preferences.URP.referralLookupOutstanding.value == true {
+                var refCode: String?
+                
+                if Preferences.URP.referralCode.value == nil {
```

Comment:
@jhreis - You missed setting the `refcode = Preferences.URP.referralCode.value` if the value is not nill.

Reply:
That's set later, when it processes the ref code, the URP class takes care of that.

Reply:
Makes sense, all good 👍